master/internal/db/postgres_cluster.go

```diff
@@ -32,6 +32,38 @@ WHERE NOT EXISTS ( SELECT * FROM cluster_id );
 	return uuidVal[0], nil
 }
 
+func (db *PgDB) UpdateClusterHeartBeat(current_cluster_heartbeat time.Time) error {
+	var uuidVal []string
+	if err := db.sql.Select(&uuidVal, `SELECT cluster_id FROM cluster_id`); err != nil {
+		return errors.Wrapf(err, "error reading cluster_id from cluster_id table")
+	}
+	if len(uuidVal) != 1 {
+		return errors.Errorf(
+			"expecting exactly one cluster_id from cluster_id table, %d values found", len(uuidVal),
+		)
+	}
+
+	_, err := db.sql.Exec(`UPDATE cluster_id SET cluster_heartbeat = $1`, current_cluster_heartbeat) //now() returns UTC time
+
+	return err
+}
+
+func (db *PgDB) GetClusterHeartBeat() (time.Time, error) {
+	var row_heartbeat []string
+	if err := db.sql.Select(&row_heartbeat, `SELECT cluster_heartbeat FROM cluster_id`); err != nil {
+		return time.Time{}, errors.Wrapf(err, "error reading cluster_heartbeat from cluster_id table")
+	}
+	if len(row_heartbeat) != 1 {
+		return time.Time{}, errors.Errorf(
+			"expecting exactly one cluster_heartbeat from cluster_id table, %d values found", len(row_heartbeat),
+		)
+	}
+
+	var layout = "2006-01-02T15:04:05.000Z"
+	cluster_heartbeat, err := time.Parse(layout, row_heartbeat[0])
+	return cluster_heartbeat, err
```

Comment:
i commented elsewhere i dont think we need this func, but i'm also curious, i would think this should just work if you use time.Time rather than string + time.Parse. is that not the case?

Reply:
Do you mean something like this: 

var rowHeartbeat time.Time 
	if err := db.sql.Select(&rowHeartbeat, `SELECT cluster_heartbeat FROM cluster_id`); err != nil {
		return time.Time{}, errors.Wrapf(err, "error reading cluster_heartbeat from cluster_id table")
	}

Yes, I guess that could work. I'm not sure I'll have to try that.